### ISSUE TITLE
It always has different hashcode even though everything is same

### ISSUE THREAD

**Issue by memishood** [avdosev/either_dart#5]
```dart
    test('should be same', () {
      final x = Left('test');
      final z = Left('test');
      expect(x, z);
    });
```

**Result:**
Expected: <Instance of 'Left<String, dynamic>'>
Actual: <Instance of 'Left<String, dynamic>'>

**Comment by avdosev** [comment]
@memishood 
You're right. This is really a problem. I published release 0.3.0 with a fix

**Comment by memishood** [comment]
It's working nicely anymore, thanks!
